bin/run-canonical-extensions-tests.sh

```
@@ -0,0 +1,106 @@
+fi
+read -r -p "Which WordPress version should we use for testing (e.g., 6.8, latest or empty to use defaults)?: " wordpress
+read -r -p "Which PHP version should we use for testing (e.g., 7.4, 8.4 or empty to use defaults)?: " php
+
+read -r -p "Which GitHub repositories needs to be tested (e.g. https://github.com/woocommerce/woocommerce, space separated list or empty to use defaults)?: " -a repositories
+if [[ ${#repositories[@]} -eq 0 ]]; then
+	# Fetch canonical extensions list.
+	file='/tmp/WOOCOMMERCE_CANONICAL_EXTENSIONS'
+	echo -n 'Fetching extensions list: ';
+	# The variable can be actualized under https://github.com/woocommerce/woocommerce/settings/variables/actions (mix of public and private repository URLs)
+	( gh variable get CANONICAL_EXTENSIONS > $file && echo 'done' ) || ( echo 'error' && exit 1 )
+	repositories=( $( cat $file | tr -d '\r' | tr '\n' ' ' ) )
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)